compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3663,21 +3663,25 @@ object Parsers {
 
     /**  ExtMethod  ::=  {Annotation [nl]} {Modifier} ‘def’ DefDef
      */
-    def extMethod(numLeadParams: Int): DefDef =
+    def extMethodOpt(numLeadParams: Int): Option[DefDef] =
       val start = in.offset
       val mods = defAnnotsMods(modifierTokens)
-      accept(DEF)
-      defDefOrDcl(start, mods, numLeadParams)
-
+      if in.token == DEF then
+        accept(DEF)
+        Some(defDefOrDcl(start, mods, numLeadParams))
+      else
+        None
+        
     /** ExtMethods ::=  ExtMethod | [nl] ‘{’ ExtMethod {semi ExtMethod ‘}’
      */
     def extMethods(numLeadParams: Int): List[DefDef] = checkNoEscapingPlaceholders {
       val meths = new ListBuffer[DefDef]
       while
-        meths += extMethod(numLeadParams)
-        statSepOrEnd(meths, what = "extension method")
+        val m = extMethodOpt(numLeadParams)
+        m.foreach(meths += _)
+        m.isDefined && statSepOrEnd(meths, what = "extension method")
```

Comment:
```suggestion
        if !m.isDefined then syntaxErrorOrIncomplete("`def` expected")
        statSepOrEnd(meths, what = "extension method")
```

I think this may fix the CI failure in tests/neg/extension-methods.scala

Reply:
Thanks for help! I played with your suggestion and found even more compact way to fix this problem